lib/valueflow.cpp

```diff
@@ -8009,7 +8016,8 @@ static void valueFlowUninit(TokenList& tokenlist, const Settings* settings)
         if (partial)
             continue;
 
-        valueFlowForward(start, tok->scope()->bodyEnd, var->nameToken(), uninitValue, tokenlist, settings);
+        if (result.terminate != Analyzer::Terminate::Modified)
```

Comment:
This seems strange. Why are we skipping forwarding only when the last variable in the list terminates as modified?

Reply:
This happens for every variable, right?
Why are there even two forwards? Anyway, the second call was setting an uninit value (for `return s`) when the MemberExpressionAnalyzer had terminated after an assignment, so I tried to prevent that.

Reply:
> This happens for every variable, right?
> Why are there even two forwards?

There is a forward for every variable in the struct. 

> Anyway, the second call was setting an uninit value (for return s) when the MemberExpressionAnalyzer had terminated after an assignment, so I tried to prevent that.

But this starts at the same spot, so why does the last `valueFlowForward` not catch the assignment?

Reply:
My thinking was, because it doesn't do the special MemberExpressionAnalyzer stuff?

Reply:
Yea thats it. And the special matching stuff is too just match the lifetimes since we point to class variables. Ideally, we should update the lifetime to point to the correct token, but this would require changing a lot of assumptions through out.

Reply:
Fixed this in #5320.